### DISCUSSION TITLE
What is unit of measure for token timeout?

### DISCUSSION THREAD

**Discussion by rzeq** [AzureAD/azure-activedirectory-library-for-python#259] [Q&A]
Referring to bellow code, what is unit of measure of timeout parameter for authentication_context?
https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/a1b9de817b83fddf9adf3a5fcd2ce71af2cb3720/adal/authentication_context.py#L109

**Comment by rayluo** [comment]
It is documented in the docs: https://adal-python.readthedocs.io/en/latest/#adal.AuthenticationContext
and in the source code: https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/1.2.7/adal/authentication_context.py#L74-L76

So, the answer is, it is the `requests`'s timeout. Its official definition is available in requests's own doc.

By the way, you really should be investing your effort on MSAL Python instead. The beginning of [ADAL Python's readme](https://github.com/AzureAD/azure-activedirectory-library-for-python#readme) described why.

**Reply by rzeq** [reply]
I cannot see where it says in docs or code if this is in seconds, milliseconds, microseconds or some other unit.

**Reply by rayluo** [reply]
The second link in my previous answer says it is a parameter being fed to Requests. So, its official definition will only be found in [Requests' doc](https://docs.python-requests.org/en/latest/).
